ui/perfherder/helpers.js

```diff
@@ -332,3 +341,201 @@ export const getGraphsLink = function getGraphsLink(seriesList, resultSets, time
 
   return `perf.html#/graphs${createQueryParams(params)}`;
 };
+
+export const httpParamSerializer = paramsObject => map(paramsObject, (value, key) => `${key}=${value}`).join('&');
+
+// old PhAlerts' inner workings
+const Alert = function Alert(alertData, optionCollectionMap) {
+  Object.assign(this, alertData);
```

Comment:
Can you make these stand-alone functions and not classes or attributes of Alert? A lot of this will probably be moving into separate components and whatever doesn't should follow the format we use in our `helpers` files.

Reply:
I agree stand-alone functions would be best. 

For anything that does end up wanting to be a class in the future, you can use [ES6 classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) since:
* all the [browsers we support](https://github.com/mozilla/treeherder/blob/9168eba8e838ed9bc8e0c8f76b0d6ce6312be040/.neutrinorc.js#L106-L111) have [native support for them](https://caniuse.com/#feat=es6-class)
* we also transpile using Babel anything that's [stage 4](https://github.com/tc39/proposals/blob/master/finished-proposals.md) plus a couple of others (eg class properties)

Reply:
I think we'll likely want to handle state at the component level. These classes currently seem quite unwieldy with methods that are only used once (for some examples, the `getSubTestsUrl` and `getIssueTrackerUrl`). Many of these methods can be made more generic, utilize existing helpers, or be moved into the components that will utilize them. What are the benefits you are seeing to keeping Alert and AlertSummary as classes?

If some of this seems a bit thorny to extract while the rest of the logic is still in controllers, I'm OK with leaving a few methods as members of a class for now, keeping in mind their form might change. This helpers file will unfortunately be messy until the conversion is finished.